Do not link the test programs against a separate Boost.System library;
use the header-only implementation from the packaged Boost version.

Index: tests/Makefile
--- tests/Makefile.orig
+++ tests/Makefile
@@ -18,7 +18,6 @@ ifneq (, $(findstring mingw, $(SYS))$(findstring windo
 endif
 
 LDLIBS = \
-	-lboost_system$(BOOST_SUFFIX) \
 	-lboost_program_options$(BOOST_SUFFIX) \
 	-lssl \
 	-lcrypto \
